File: Python/0055_jump_game/0055_jump_game.py
class Solution:
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        maximum = nums[0]
        index = 0
        while index <= maximum and index < len(nums):
            if index + nums[index] > maximum:
                maximum = index + nums[index]
            index += 1
        return True if maximum >= len(nums) - 1 else False

File: Cplusplus/0055_jump_game/0055_jump_game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int maximum = nums[0];
        int index = 0;
        while (index <= maximum && index < nums.size()) {
            if (index + nums[index] > maximum)
                maximum = index + nums[index];
            index++;
        }
        return maximum >= nums.size() - 1 ? true : false;
    }
};
